pyaedt/modeler/PrimitivesTwinBuilder.py

```diff
@@ -356,3 +356,73 @@ def create_pnp(self, compname=None, location=[], angle=0, use_instance_id_netlis
         )
 
         return id
+
+    @aedt_exception_handler
+    def create_periodic_waveform_source(
+        self,
+        compname=None,
+        type="SINE",
+        amplitude=100,
+        freq=50,
+        phase=0,
+        offset=0,
+        delay=0,
+        location=[],
+        angle=0,
+        use_instance_id_netlist=False,
+    ):
+        """
+        Create a periodic waveform source (non conservative real output).
+
+        Parameters
+        ----------
+        compname : str, optional
```

Comment:
@SushobhitGupt Should we rename this argument ``source_name`` as it is the name of the voltage source. I understand that we are creating a component behind the scene. It is just to be sure.

Reply:
@MaxJPRey compname is a known Twin Builder terminology, we probably would not want to change args based on what type of component is being placed (like sources in this case). keeping it same would keep consistency with other components defined in the file

Reply:
@SushobhitGupt It makes sense.